core/include/drivers/ls_sfp.h

```diff
@@ -0,0 +1,181 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright 2022 NXP
+ *
+ * Definitions for the LS-series Security Fuse Processor (SFP) driver.
+ *
+ */
+
+#ifndef __DRIVERS_LS_SFP_H
+#define __DRIVERS_LS_SFP_H
+
+#include <drivers/ls_gpio.h>
+#include <stdlib.h>
+#include <tee_api_types.h>
+
+/* FDT property names */
+#define FDT_PROP_GPIO "povdd-gpio-controller"
+#define FDT_PROP_PIN "povdd-gpio-pin"
+
+/* SFP instruction register */
+/* SFP is big endian */
+#define SFP_INGR_PROGFB_CMD 0x2
+#define SFP_INGR_ERROR_MASK 0x100
+
+/* SFP configuration register */
+#define SFP_SFPCR_SB 0x20000000
+
+/* SFP OEM security policy register 0 */
+#define SFP_OSPR0_WP 0x1
+#define SFP_OSPR0_ITS 0x4
+
+/* SFP OEM security policy register 1 */
+#define SFP_OSPR1_DBLEV_MASK 0x7
+#define SFP_OSPR1_DBLEV_OPEN 0x0
+#define SFP_OSPR1_DBLEV_CLOSED_NOTIFY 0x1
+#define SFP_OSPR1_DBLEV_CLOSED_SILENT 0x2
+#define SFP_OSPR1_DBLEV_CLOSED 0x4
+
+/**
+ * struct ls_sfp_data - Compact data struct of all SFP registers.
+ * @ingr:		Instruction Register.
+ * @svhesr:		Secret Value Hamming Error Status Register.
+ * @sfpcr:		SFP Configuration Register.
+ * @version:		SFP Version Register.
+ * @ospr0:		OEM Security Policy Register 0.
+ * @ospr1:		OEM Security Policy Register 1.
+ * @dcvr0:		Debug Challenge Value Register 0.
+ * @dcvr1:		Debug Challenge Value Register 1.
+ * @drvr0:		Debug Response Value Register 0.
+ * @drvr1:		Debug Response Value Register 1.
+ * @fswpr:		Factory Section Write Protect Register.
+ * @fuidr0:		Factory Unique ID Register 0.
+ * @fuidr1:		Factory Unique ID Register 1.
+ * @isbccr:		ISBC Configuration Register.
+ * @fspfr[0x3]:		Factory Scratch Pad Fuse Registers.
+ * @otpmkr[0x8]:	One Time Programmable Master Key.
+ * @srkhr[0x8]:		Super Root Key Hash Register.
+ * @ouidr[0x5]:		OEM Unique ID/Scratch Pad Fuse Registers.
+ *
+ */
+struct ls_sfp_data {
+	uint32_t ingr;
+	uint32_t svhesr;
+	uint32_t sfpcr;
+	uint32_t version;
+	uint32_t ospr0;
+	uint32_t ospr1;
+	uint32_t dcvr0;
+	uint32_t dcvr1;
+	uint32_t drvr0;
+	uint32_t drvr1;
+	uint32_t fswpr;
+	uint32_t fuidr0;
+	uint32_t fuidr1;
+	uint32_t isbccr;
+	uint32_t fspfr[0x3];
+	uint32_t otpmkr[0x8];
+	uint32_t srkhr[0x8];
+	uint32_t ouidr[0x5];
+};
+
+#define SFP_DATA_SIZE		sizeof(struct ls_sfp_data)
+#define SFP_FSPFR_SIZE		sizeof(((struct ls_sfp_data *)0)->fspfr)
+#define SFP_OTPMKR_SIZE		sizeof(((struct ls_sfp_data *)0)->otpmkr)
+#define SFP_SRKHR_SIZE		sizeof(((struct ls_sfp_data *)0)->srkhr)
+#define SFP_OUIDR_SIZE		sizeof(((struct ls_sfp_data *)0)->ouidr)
```

Comment:
I don't think these macros are super useful. Callers can as well write `sizeof(*sfp_date_ptr)` or `ARRAY_SIZE(sfp_data->fspfr)` etc. which IMO are much easier to read.